apps/studio/components/ui/SqlWarningAdmonition.tsx

```diff
@@ -7,32 +7,47 @@ export interface SqlWarningAdmonitionProps {
   onConfirm: () => void
   disabled?: boolean
   className?: string
+  /** Optional override primary message */
+  message?: string
+  /** Optional override secondary message */
+  subMessage?: string
+  /** Optional override labels */
+  cancelLabel?: string
+  confirmLabel?: string
 }
 
-const SqlWarningAdmonition = ({
+export const SqlWarningAdmonition = ({
   warningType,
   onCancel,
   onConfirm,
   disabled = false,
   className,
+  message,
+  subMessage,
+  cancelLabel,
+  confirmLabel,
 }: SqlWarningAdmonitionProps) => {
   return (
     <Admonition
       type="warning"
       className={`mb-0 rounded-none border-0 shrink-0 bg-background-100 ${className}`}
     >
       <p className="text-xs !mb-1">
-        {warningType === 'hasWriteOperation'
-          ? 'This query contains write operations.'
-          : 'This query involves running a function.'}{' '}
-        Are you sure you want to execute it?
+        {message ?? (
+          <>
```

Comment:
could do a string interpolation instead of fragment here

Reply:
the message conditional could also be wrapping the `p` element? otherwise it'll be an empty `p` if there's no message